portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -35,14 +38,26 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     String message = request.getParameter("user-comment");
     long timestamp = System.currentTimeMillis();
 
-    if(name.length() != 0 && message.length() != 0){
+    if (name.length() != 0 && message.length() != 0) {
         Entity taskEntity = new Entity("Comment");
+
         taskEntity.setProperty("name", name);
         taskEntity.setProperty("message", message);
+        taskEntity.setProperty("score", getSentimentScore(message));
         taskEntity.setProperty("timestamp", timestamp);
         datastore.put(taskEntity);
     }
 
     response.sendRedirect("/index.html#comments-panel");
   }
+
+  public double getSentimentScore(String message) throws IOException {
+    Document doc = Document.newBuilder().setContent(message).setType(Document.Type.PLAIN_TEXT).build();
+    LanguageServiceClient languageService = LanguageServiceClient.create();
```

Comment:
So sometimes, creating a client to an API can be expensive. You usually only do this once when the server is starting up. Doing it per request can be laggy. Probably call `LanguageServiceClient.create();` at the beginning of the main function when the server starts up.

Reply:
I will look into that, thank you for the advice

Reply:
I sent the commit for this change in the latest branch instead of this one by accident